gammapy/cube/tests/test_psf_map.py

```diff
@@ -182,3 +193,52 @@ def test_psfmap_stacking():
 
 
 # TODO: add a test comparing make_mean_psf and PSFMap.stack for a set of observations in an Observations
+
+
+def get_npred_map():
+    position = SkyCoord(0.0, 0.0, frame="galactic", unit="deg")
+    energy_axis = MapAxis.from_bounds(
+        0.2, 10, nbin=30, unit="TeV", name="energy", interp="log"
+    )
+
+    exposure = Map.create(
+        binsz=0.02,
+        map_type="wcs",
+        skydir=position,
+        width="2 deg",
+        axes=[energy_axis],
+        coordsys="GAL",
+        unit="cm2 s",
+    )
+
+    spatial_model = GaussianSpatialModel(
+        "0 deg", "0 deg", sigma="0.2 deg", frame="galactic"
+    )
+    spectral_model = PowerLawSpectralModel(amplitude="1e-11 cm-2 s-1 TeV-1")
+    skymodel = SkyModel(spatial_model=spatial_model, spectral_model=spectral_model)
+
+    exposure.data = 1e14 * np.ones(exposure.data.shape)
+    evaluator = MapEvaluator(model=skymodel, exposure=exposure)
+
+    npred = evaluator.compute_npred()
+    return evaluator, npred
+
+
+def test_sample_coord():
+    psf_map = make_test_psfmap(0.1 * u.deg, shape="gauss")
+    eval, npred = get_npred_map()
+
+    nmap = WcsNDMap(geom=eval.geom, data=npred.data)
+    coords = nmap.sample_coord(n_events=2, random_state=0)
+
+    coords_corrected = PSFMap.sample_coord(psf_map=psf_map, map_coord=coords)
```

Comment:
Note that the way `.sample_coord()` is used is actually by directly calling `psf_map.sample_coord()`, instead of calling `PSFMap.sample_coord()` and passing the `psf_map`.